### PR TITLE
Update deployment documentation for Vercel

### PR DESCRIPTION
**Pull request by leerob**

### CODE REVIEW

**Comment by Tobbe** [comment]
Wow! Love how much of the instructions you could remove. Thanks for the updates

**Comment by ajcwebdev** [comment]
This is killer Lee! One minor change I would recommend is removing the `$` symbol from the terminal commands and separating out the commands from the terminal output. The docs are setup to have copy/paste buttons on the commands for readers who are following along.

Not related to this PR but I also noticed another example of this in section [3 - Prisma and Database](https://redwoodjs.com/docs/deploy#3-prisma-and-database).

File: docs/deploy.md [hunk 1]
```diff
@@ -105,71 +105,32 @@ If you simply want to experience the Render deployment process, including a Post
...
-#### Step 1. Serverless Functions Path
-Run the following CLI Command:
-```shell
-yarn rw setup deploy vercel
+```bash
+$ npm i -g vercel
+$ vercel init redwoodjs
+Vercel CLI
+> Success! Initialized "redwoodjs" example in ~/your-folder.
+- To deploy, `cd redwoodjs` and run `vercel`.
```
Comment (leerob) [inline docs/deploy.md:120]:
```suggestion
npm i -g vercel
vercel init redwoodjs
```

**Comment by leerob** [comment]
> One minor change I would recommend is removing the $ symbol from the terminal commands and separating out the commands from the terminal output.

Agreed, done!

> Also, is this meant to replace yarn rw setup deploy vercel entirely, or would there be a reason to use one method over the other?

The `vercel init` creates a new Redwood project based on [this template](https://github.com/vercel/vercel/tree/main/examples/redwoodjs). I am unsure the differences between that and `yarn rw setup deploy vercel`.

**Comment by ajcwebdev** [comment]
> The `vercel init` creates a new Redwood project based on this template. I am unsure the differences between that and `yarn rw setup deploy vercel`.

Okay, so `vercel init` is for generating a new Redwood app configured for Vercel, whereas `yarn rw setup deploy <provider>` configures an existing Redwood project for a specific provider (in this case Vercel). This usually involves a few modifications such as:
* Setting the URL for the API (For Vercel the `apiUrl` is set to `/api`).
* Generating a vendor specific configuration as code file such as `netlify.toml` or `serverless.yml`

We'll want both instructions because they cover fairly separate use cases.
* `vercel init` - Wants to start a new Redwood project and already knows they will be using Vercel
* `setup deploy vercel` - Wants to start without a chosen provider (which is the default if you use Create Redwood App) and then decides later they want to use Vercel

**Comment by thedavidprice** [comment]
Quick check in: this is on my priority list! I do have some direction/suggestions for discussion. 

Will loop back after getting release out the door. Thanks for your help + patience @leerob

**Comment by leerob** [comment]
To check my understanding: `setup deploy vercel` would configure your _existing_ project to use Vercel?

**Comment by ajcwebdev** [comment]
Correct. When creating a boilerplate Redwood app you would run a command like:

```bash
yarn create redwood-app redwood-vercel
```

The default `redwood.toml` configuration will include the following:

```bash
[web]
  title = "Redwood App"
  port = 8910
  apiUrl = "/.redwood/functions"
  includeEnvironmentVariables = []
[api]
  port = 8911
[browser]
  open = true
```

From here you can pick any of the current setup deployment options and they will either modify your `redwood.toml` configuration file, create a new configuration as code file for a specific provider, or both.

For Vercel, you would run:

```bash
yarn rw setup deploy vercel
```

This will modify your `redwood.toml` configuration to the following:

```diff
[web]
  title = "Redwood App"
  port = 8910
- apiUrl = "/.redwood/functions"
+ apiUrl = "/api"
  includeEnvironmentVariables = []
[api]
  port = 8911
[browser]
  open = true
```

Here's an example repo using the latest Redwood version for reference: [`redwood-vercel`](https://github.com/ajcwebdev/redwood-vercel)
Deployed site: [redwood-vercel.vercel.app/](https://redwood-vercel.vercel.app/)

**Review by thedavidprice — CHANGES_REQUESTED** [review]
@leerob Huge thank you for getting this rolling (and apologies for delayed review). In general, this prompted me to outline some needed improvement for this doc overall. (Just an FYI. See https://github.com/redwoodjs/redwoodjs.com/issues/972)

I understand the approach you took here, but for this doc we want to focus on first using the applicable `yarn redwood setup deploy <provider>` command along with the "push to git repo" workflow. After that, we welcome covering additional topics or linking to a Vercel deployment doc.

Some ideas for discussion about how to proceed:
- what if devs start with `yarn rw setup deploy vercel`, which could include adding the `vercel` package as part of the command?
- Typically, we'd want to first highlight the "Vercel for Git" steps. But I do like how simple the Vercel CLI is. Again, highlighting both is great, it's just how we want to emphasize/prioritize. TBD
    - Note: adding a `DATABASE_URL` is not optional (at least for 99% of apps)
- What about other Vercel deploy docs to link to?

I took a look at this doc on the Vercel site:
https://vercel.com/guides/deploying-redwood-with-vercel

Curious how you're managing the template:
1. latest versions?
2. DB?
3. we have a really nice splash page, why not use the default?

We'd love to help this make Redwood + Vercel shine a bit more. Let's discuss!

**Comment by leerob** [comment]
> what if devs start with yarn rw setup deploy vercel, which could include adding the vercel package as part of the command?

That's fine with me - unsure if it needs to install vercel, since I think most folks install it globally.

> Typically, we'd want to first highlight the "Vercel for Git" steps. But I do like how simple the Vercel CLI is. Again, highlighting both is great, it's just how we want to emphasize/prioritize.

We can switch the order, whichever works 😄 

> Note: adding a DATABASE_URL is not optional (at least for 99% of apps)

Can remove optional!

**Comment by leerob** [comment]
> Curious how you're managing the template:

1. We periodically update to the latest versions
2. No preference for database, remaining unopinionated here
3. Unsure what you mean about this page - we're using the default output from making a new Redwood app

**Comment by jtoar** [comment]
Closing this as I've moved this PR over to the main repo where the docs now live: https://github.com/redwoodjs/redwood/pull/5195.

**Pull request closed without merging**